### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by Polber**
Update release workflow to use modular jobs similar to Java PR workflow. Also includes site report for easier failure tracking in release and scheduled Java PR workflows.

### CODE REVIEW

**Comment by Polber** [comment]
@damccorm Could you PTAL when you have a chance? These changes should help alleviate pain points in the release and when examining test failures on scheduled Java PR's (and the release).

**Comment by damccorm** [comment]
> @damccorm Could you PTAL when you have a chance? These changes should help alleviate pain points in the release and when examining test failures on scheduled Java PR's (and the release).

Yes - though moving forward, could you try to request review only once the PR is ready/checks are passing? It is pretty hard to follow notifications otherwise.

Same applies to https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1697 for example - I got the review requested notification, but checks were initially failing then there was a series of commits and I'm not sure if you want a review.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,99 +21,195 @@ permissions:
...
-    - name: Push tags
-      run: |
-        git push -u origin --tags
-    - name: Release
-      run: |
-        gh release create $TAG_NAME --title "Dataflow Templates $TAG_NAME" --notes ""
-      env:
-        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        TAG_NAME: ${{ steps.variables.outputs.tagName }}
+      - name: Get releaser identity
```
Comment (damccorm) [inline .github/workflows/release.yml:31]: Looks like there was an indentation change, which I think may break this workflow - could we revert? It will also make this section much easier to review

Comment (Polber) [reply .github/workflows/release.yml]: Indentation change fixes format to match standard YAML format and matches that of the other workflows

https://yaml.org/spec/history/2003-09-01.html#:~:text=between%20YAML%20systems.-,5.1%20Sequence,-This%20type%20family

I reverted for easier review, but will probably re-indent before submit unless you have concerns.

**Review by damccorm — COMMENTED** [review]
Overall LGTM, thanks for doing this. Just had a few comments

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -21,9 +21,12 @@ permissions:
   contents: write
 
 jobs:
-  release:
+  create_release_branch:
     name: Create Release
-    runs-on: [self-hosted, release]
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (Polber) [reply .github/workflows/release.yml:26]: This particular workflow does not need a special machine, so I set it to ubuntu-latest to avoid using a self-hosted machine when not necessary, but I can set to self-hosted if you think that makes more sense.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -56,64 +59,157 @@ jobs:
...
-        gh release create $TAG_NAME --title "Dataflow Templates $TAG_NAME" --notes ""
-      env:
-        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        TAG_NAME: ${{ steps.variables.outputs.tagName }}
+        RELEASE_BRANCH_NAME: ${{ steps.variables.outputs.release_branch_name }}
+  java_unit_tests:
+    name: Unit Tests
+    needs: create_release_branch
+    timeout-minutes: 60
+    runs-on: [ self-hosted, release ]
```
[the remark this answers is not in the document]
Comment (Polber) [reply .github/workflows/release.yml:72]: Good point, it does not. I can add some more

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -56,64 +59,157 @@ jobs:
...
+          token: ${{ secrets.RELEASE_TOKEN }}
+      - name: Create artifacts and push
+        run: |
+          ./cicd/run-release \
+            --it-project="dataflow-templates" \
+            --release-bucket-name="dataflow-templates-staging" \
+            --release-libraries-bucket-name="dataflow-templates-libraries" \
+            --release-stage-prefix="${CANDIDATE_NAME}"
+        env:
+          CANDIDATE_NAME: ${{ inputs.candidateName }}
```
[the remark this answers is not in the document]
Comment (Polber) [reply .github/workflows/release.yml:181]: Sure I can change it back

**Comment by Polber** [comment]
@damccorm On second thought, I am leaving the majority of the release as it was, and instead am just replacing the JUnit report step with the newly-added Site Report step in order to generate a more meaningful test result report upon release failures.

The other changes were not beneficial enough to outweigh added cost for machines and added code maintenance

**Review by damccorm — APPROVED** [review]
I actually liked the other changes - we can take this change for now, but breaking the jobs up does generally seem beneficial
